core/aclmgmt/resourceprovider.go

```diff
@@ -138,6 +141,11 @@ func (rp *aclmgmtPolicyProviderImpl) CheckACL(polName string, idinfo interface{}
 	return nil
 }
 
+// This function should not be called because CheckACLNoChannel does not support overriding default provider
+func (rp *aclmgmtPolicyProviderImpl) CheckACLNoChannel(resName string, idinfo interface{}) error {
+	return fmt.Errorf("overriding default acl provider is not supported")
+}
+
```

Comment:
This really looks bad... an implementation knows the behavior of the consumer (which is usually the other way around). I am not well versed with this package but is it not possible for this impl to implement a separate interface which does not have this function - or rather does it have to implement this interface at all? I see this getting used only at one place in the production code where it does not need to implement this interface.

Reply:
Removed because peer resource does not allow override and never gets here.